savant/deepstream/pyfunc.py

```diff
@@ -40,7 +40,7 @@ def get_cuda_stream(self, frame_meta: NvDsFrameMeta):
         """Get a CUDA stream that can be used to asynchronously process a frame in a batch.
         All frame CUDA streams will be waited for at the end of batch processing.
         """
-        if frame_meta.batch_id not in self.frame_streams[frame_meta.batch_id]:
+        if frame_meta.batch_id not in self.frame_streams:
```

Comment:
Please add debug batch_id to ensure it works properly on batched operations.

Reply:
Added debug logging.

Tested `opencv_cuda_bg_remover_mog2` sample on 2 sources with batch_size=1 and batch_size=2.